File: src/main/java/com/github/mgramin/sqlboot/model/uri/wrappers/SqlPlaceholdersWrapper.java
/**
 * The MIT License (MIT)
 *
 * Copyright (c) 2016-2017 Maksim Gramin
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.github.mgramin.sqlboot.model.uri.wrappers;

import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;
import com.github.mgramin.sqlboot.model.uri.Uri;

/**
 * @author Maksim Gramin (mgramin@gmail.com)
 * @version $Id$
 * @since 0.1
 */
public class SqlPlaceholdersWrapper implements Uri {

    private final Uri origin;

    public SqlPlaceholdersWrapper(final Uri origin) {
        this.origin = origin;
    }

    @Override
    public String type() {
        return origin.type();
    }

    @Override
    public String action() {
        return origin.action();
    }

    @Override
    public List<String> path() {
        return origin.path().stream()
                .map(v -> v.replace("*", "%"))
                .collect(Collectors.toList());
    }

    @Override
    public Boolean recursive() {
        return origin.recursive();
    }

    @Override
    public Map<String, String> params() {
        return origin.params();
    }

    @Override
    public Map<String, String> filters() {
        return origin.filters();
    }

}

File: src/main/kotlin/com/github/mgramin/sqlboot/model/uri/wrappers/SqlPlaceholdersWrapper.kt
/*
 * BSD 3-Clause License
 *
 * Copyright (c) 2019, CROC Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * 1. Redistributions of source code must retain the above copyright notice, this
 *    list of conditions and the following disclaimer.
 *
 * 2. Redistributions in binary form must reproduce the above copyright notice,
 *    this list of conditions and the following disclaimer in the documentation
 *    and/or other materials provided with the distribution.
 *
 * 3. Neither the name of the copyright holder nor the names of its
 *    contributors may be used to endorse or promote products derived from
 *    this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

package com.github.mgramin.sqlboot.model.uri.wrappers

import com.github.mgramin.sqlboot.model.uri.Uri

/**
 * @author Maksim Gramin (mgramin@gmail.com)
 * @version $Id: bce1e45321cd26b04727a56fc64fbc4ff54bd82c $
 * @since 0.1
 */
class SqlPlaceholdersWrapper(private val origin: Uri) : Uri {

    override fun type() = origin.type()

    override fun path() = origin.path().map { v -> v.replace("*", "%25") }.toList()

    override fun path(index: Int) = origin.path(index).replace("*", "%25")

    override fun params() = origin.params().map { it.key to it.value.replace("*", "%25") }.toMap()

    override fun action() = origin.action()

    override fun connection() = origin.connection()

    override fun toString() = origin.toString()

}
